api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java

```diff
@@ -402,29 +402,83 @@ public static String getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreA
+
     String query =
-        "SELECT final.patient_id FROM( "
-            + " SELECT c.patient_id, MAX(c.encounter_date) as encounter_date FROM( "
-            + " SELECT p.patient_id, MAX(e.encounter_datetime) AS encounter_date FROM patient p INNER JOIN encounter e "
-            + " ON p.patient_id=e.patient_id WHERE p.voided=0 AND e.voided=0 AND e.location_id=:location AND "
-            + " e.encounter_datetime <=:onDate AND e.encounter_type=%d GROUP BY p.patient_id "
-            + " UNION  "
-            + " SELECT p.patient_id, MAX(o.value_datetime) AS encounter_date FROM patient p INNER JOIN encounter e "
-            + " ON p.patient_id=e.patient_id INNER JOIN obs o ON o.encounter_id=e.encounter_id WHERE p.voided=0 AND e.voided=0 "
-            + " AND o.voided=0 AND e.encounter_type=%d AND o.concept_id=%d AND o.value_datetime IS NOT NULL AND "
-            + "e.location_id=:location AND e.encounter_datetime<=:onDate GROUP BY p.patient_id "
-            + " ) c GROUP BY c.patient_id "
-            + " ) final WHERE DATE_ADD(final.encounter_date, INTERVAL 90 DAY) < :onDate ";
-    return String.format(
-        query, pharmacyEncounterType, mastercardEncounterType, drugPickupConceptId);
+        "SELECT final.patient_id   "
+            + "             from(   "
+            + "                 SELECT   "
+            + "                     most_recent.patient_id, Date_add(Max(most_recent.value_datetime), interval 60 day) final_encounter_date   "
+            + "                 FROM   (SELECT fila.patient_id, o.value_datetime from (  "
+            + "                             SELECT enc.patient_id,   "
+            + "                                 Max(enc.encounter_datetime)  encounter_datetime  "
+            + "                             FROM   patient pa   "
+            + "                                 inner join encounter enc   "
+            + "                                     ON enc.patient_id =  pa.patient_id   "
+            + "                                 inner join obs obs   "
+            + "                                     ON obs.encounter_id = enc.encounter_id   "
+            + "                             WHERE  pa.voided = 0   "
+            + "                                 AND enc.voided = 0   "
+            + "                                 AND obs.voided = 0   "
+            + "                                 AND obs.concept_id =  ${returnVisitDateForArvDrugConcept}   "
+            + "                                 AND obs.value_datetime IS NOT NULL   "
+            + "                                 AND enc.encounter_type = ${arvPharmaciaEncounterType}   "
+            + "                                 AND enc.location_id = :location   "
+            + "                                 AND enc.encounter_datetime <= :onDate   "
+            + "                             GROUP  BY pa.patient_id) fila   "
+            + "                         INNER JOIN encounter e on  "
```

Comment:
This join does not seem necessary, what is it for?

Reply:
this is needed to  other attribuites of  encounters  table and join with maximum encounters